src/validation.cpp

```diff
@@ -4345,6 +4349,10 @@ bool ChainstateManager::AcceptBlockHeader(const CBlockHeader& block, BlockValida
             LogDebug(BCLog::VALIDATION, "%s: Consensus::CheckBlockHeader: %s, %s\n", __func__, hash.ToString(), state.ToString());
             return false;
         }
+        if (!CheckHeaderNotFuture(block, state)) {
+            LogDebug(BCLog::VALIDATION, "%s: Consensus::CheckHeaderNotFuture: %s, %s\n", __func__, hash.ToString(), state.ToString());
```

Comment:
Why is this not `LogError`?

Reply:
To keep the existing level for this error. I suppose the existing level is debug due to log-filling concerns, maybe with very-low-difficulty headers?

Thanks for pointing this out though, i double checked the other log line in `TestBlockValidity` and there it should have been `LogError`. Fixed.

Reply:
Someone else making an invalid block is not a reason for our node to shut down, so `LogError` would be wrong.

We also run this before the actual proof-of-work check, and headers can be sent repeatedly, so need to be careful about log spam.

You can however drop `__func__`, because we have `-logsourcelocations`.